src/update_handler.c

```diff
@@ -1141,6 +1142,123 @@ static gboolean img_to_fs_handler(RaucImage *image, RaucSlot *dest_slot, const g
 	return res;
 }
 
+static gboolean img_to_boot_mbr_switch_handler(RaucImage *image, RaucSlot *dest_slot, const gchar *hook_name, GError **error)
+{
+	gboolean res = FALSE;
+	int out_fd;
+	g_autoptr(GUnixOutputStream) outstream = NULL;
+	GFile *destslotfile = NULL;
+	GError *ierror = NULL;
+	struct mbr_switch_partition dest_partition;
+
+	/* run slot pre install hook if enabled */
+	if (hook_name && image->hooks.pre_install) {
+		res = run_slot_hook(hook_name, R_SLOT_HOOK_PRE_INSTALL, NULL,
+				dest_slot, &ierror);
+		if (!res) {
+			g_propagate_error(error, ierror);
+			goto out;
+		}
+	}
+
+	if (!dest_slot->region_start || !dest_slot->region_size) {
+		g_set_error(error, R_UPDATE_ERROR, R_UPDATE_ERROR_FAILED,
+				"Region configuration must be provided for type boot-mbr-switch");
+		res = FALSE;
+		goto out;
+	}
+
+	res = r_mbr_switch_get_inactive_partition(dest_slot->device, &dest_partition,
+			dest_slot->region_start, dest_slot->region_size,
+			&ierror);
+	if (!res) {
+		g_propagate_error(error, ierror);
+		goto out;
+	}
+
+	g_message("Found inactive boot partition at %luB, size %luB",
+			dest_partition.start, dest_partition.size);
+
+	if (dest_partition.size != image->checksum.size) {
```

Comment:
We could easily handle image-size < partition size as well. Do you have any reason do be more strict here?